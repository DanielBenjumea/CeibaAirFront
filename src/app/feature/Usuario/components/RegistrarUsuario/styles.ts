import styled from 'styled-components';

export const SpanError = styled.span`
  color: #f62d2d;
  font-weight: bold;
  font-size: 8px;
`;

export const FlexContainer = styled.span`
    height: 60vh;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: #F0F0F0;
`;

export const Form = styled.form`
    height: 75%;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    padding: 40px;
    background-color: #ffffff;
`;