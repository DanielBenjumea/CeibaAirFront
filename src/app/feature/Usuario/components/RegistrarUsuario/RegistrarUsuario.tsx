import * as React from 'react';
import * as Yup from 'yup';
import { FlexContainer, Form, SpanError } from './styles';
import { Button } from 'app/shared/components/Button';
import { Input } from 'app/shared/components/Input';
import Swal from 'sweetalert2';
import { axiosIntance } from 'app/core/config/AxiosConfig';
import { useFormik } from 'formik';
import { useHistory } from 'react-router-dom';

interface FormValues {
	nombre: string;
	clave: string;
}

const validationSchema = Yup.object().shape<FormValues>({
	nombre: Yup.string().required('El campo nombre es requerido.'),
	clave: Yup.string().required('El campo password es requerido.')
});

export const FormRegistrarusuario = () => {
	const history = useHistory();
	const initialValues = {
		nombre: '',
		clave: ''
	};

	const handleSubmit = async (values: FormValues) => {
		try {
			const response = await axiosIntance.post('/usuarios', {
				nombre: values.nombre,
				clave: values.clave,
				fechaCreacion: new Date()
			});

			if (response.status === 201) {
				Swal.fire(
					'Usuario Creado',
					'Su usuario fue creado satisfactoriamente / serÃ¡ redirigido a la pantalla inicial',
					'success'
				).then(() => {
					history.push('/');
				});
			}
		} catch (error) {
			Swal.fire(
				'Error al crear usuario',
				'Nombre de usuario ya existe',
				'error'
			);
		}
	};

	const formik = useFormik({
		initialValues,
		validationSchema,
		onSubmit: handleSubmit
	});

	return (
		<FlexContainer>
			<Form onSubmit={formik.handleSubmit}>
				<h2>Registrar usuario</h2>
				<Input
					name="nombre"
					placeholder="Nombre de usuario"
					value={formik.values.nombre}
					onChange={formik.handleChange}
				/>
				{formik.touched.nombre && formik.errors.nombre && <SpanError>{formik.errors.nombre}</SpanError>}
				<Input
					name="clave"
					placeholder="Password"
					type="password"
					value={formik.values.clave}
					onChange={formik.handleChange}
				/>
				{formik.touched.clave && formik.errors.clave && <SpanError>{formik.errors.clave}</SpanError>}

				<Button type="submit">Registrar</Button>
			</Form>
		</FlexContainer>
	);
};
