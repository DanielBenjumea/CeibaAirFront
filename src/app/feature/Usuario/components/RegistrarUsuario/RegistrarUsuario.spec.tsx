import * as React from 'react';
import { fireEvent, render, RenderResult, wait } from '@testing-library/react';
import { FormRegistrarusuario } from './RegistrarUsuario';

describe('RegistrarUsuario test', () => {
	let componentWrapper: RenderResult;

	beforeEach(() => {
		componentWrapper = render(<FormRegistrarusuario />);
	});

	it('Should match snapshot', () => {
		expect(componentWrapper.container).toMatchSnapshot();
	});

	it('Should fail on submit all fields missing', async () => {
		const elem = componentWrapper.container;
		const submitButton = elem.querySelector('button[type="submit"]');
		await wait(() => {
			submitButton && fireEvent.click(submitButton);
		});

        const spans = elem.querySelectorAll('span');

        console.log(spans);
        
        expect(spans.length).toBe(2);
	});
});
