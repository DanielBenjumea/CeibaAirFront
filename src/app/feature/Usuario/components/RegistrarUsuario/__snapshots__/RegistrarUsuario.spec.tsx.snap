// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrarUsuario test Should match snapshot 1`] = `
<div>
  <span
    class="sc-AxirZ lbJqCm"
  >
    <form
      class="sc-AxiKw hTFskS"
    >
      <h2>
        Registrar usuario
      </h2>
      <input
        class="sc-AxhUy fHwnCS"
        name="nombre"
        placeholder="Nombre de usuario"
        value=""
      />
      <input
        class="sc-AxhUy fHwnCS"
        name="clave"
        placeholder="Password"
        type="password"
        value=""
      />
      <button
        class="sc-AxhCb XVgvt"
        type="submit"
      >
        Registrar
      </button>
    </form>
  </span>
</div>
`;
